vectordatasource/transform.py

```diff
@@ -4525,3 +4584,54 @@ def backfill_from_other_layer(ctx):
                 props[layer_key] = value
 
     return layer
+
+
+def drop_layer(ctx):
+    """
+    Drops the named layer from the list of layers.
+    """
+
+    layer_to_delete = ctx.params.get('layer')
+    found_idx = None
+
+    for idx, feature_layer in enumerate(ctx.feature_layers):
+        layer_datum = feature_layer['layer_datum']
+        layer_name = layer_datum['name']
+
+        if layer_name == layer_to_delete:
+            found_idx = idx
+            break
+
+    if found_idx is not None:
+        del ctx.feature_layers[found_idx]
```

Comment:
Ah, good point, thanks! I had thought that it wasn't allowed to modify a list while iterating over it, but I think we get away with it here since we exit the loop immediately. Fixed in 7379c6a.